src/ViewModel/PlayingSongViewModel.cpp

```diff
@@ -13,15 +14,24 @@ PlayingSongViewModel::PlayingSongViewModel(QObject* parent) : QObject{parent}, p
                      &PlayingSongViewModel::playSong);
     QObject::connect(NextUpViewModel::getInstance(), &NextUpViewModel::playingSongChanged, this,
                      &PlayingSongViewModel::setPlayingSong);
-    // TODO: connect signals and slots
+
+    connect(player,&QMediaPlayer::positionChanged,this,&PlayingSongViewModel::onMusicPositionChanged);
+    connect(player,&QMediaPlayer::mediaStatusChanged,this,&PlayingSongViewModel::onMediaStatusChanged);
+    connect(player,&QMediaPlayer::playingChanged,this,&PlayingSongViewModel::setPlaying);
+    connect(player,&QMediaPlayer::playbackStateChanged,this,&PlayingSongViewModel::onPlayStateChanged);
 }
 
 PlayingSongViewModel::~PlayingSongViewModel() {
     save();
 }
 
 PlayingSongViewModel* PlayingSongViewModel::create(QQmlEngine*, QJSEngine*) {
-    return new PlayingSongViewModel();
+    return getInstance();
```

Comment:
Call `QQmlEngine::setObjectOwnership` to set correct ownership, otherwise the object will be freed twice, causing crash when quitting the application.